eZ/Publish/Core/Repository/UserService.php

```diff
@@ -576,15 +576,24 @@ public function loadUserByCredentials($login, $password)
         usleep(mt_rand(0, 30000));
 
         $spiUser = $this->userHandler->loadByLogin($login);
-        $passwordHash = $this->createPasswordHash(
-            $login,
-            $password,
-            $this->settings['siteName'],
-            $spiUser->hashAlgorithm
-        );
 
-        if ($spiUser->passwordHash !== $passwordHash) {
-            throw new NotFoundException('user', $login);
+        if ($spiUser->hashAlgorithm == User::PASSWORD_BCRYPT) {
```

Comment:
We should refactor to add a verifyPassword() function to avoid hardcoding things here, might be more `password_verify` based passwords in the future. To avoid duplications it can reuse createPasswordHash for anything but PASSWORD_BCRYPT. You can also move `usleep(mt_rand(0, 30000));` into it as it will not be needed with `password_verify` based passwords as it is ["safe against timing attacks"](http://php.net/manual/en/function.password-verify.php).